test/optimization/test_readme_sample.py

```diff
@@ -64,6 +65,9 @@ def print(*args):
         # Fix node 0 to be 1 to break the symmetry of the max-cut solution
         problem.linear_constraint([1, 0, 0, 0], '==', 1)
 
+        # Fix the random seed of SPSA (optional)
+        aqua_globals.random_seed = np.random.default_rng(123)
+
```

Comment:
The code inside the lines marked with ------------- is an exact copy of the sample in the readme.md in this repo for the purpose of ensuring the readme sample continues to work as code in repo is evolved. Can you either move the code outside of those marked lines in this test file to set the seed earlier (in which case the sample code may fail for users if they try it) or change the readme sample too so it matches what is inside those lines. Now the seed that is being set should be a number not an instance of a generator i.e. `aqua_globals.random_seed = 123` . And I will note that this seeds only Aqua and SPSA as you note and does not do so for qasm simulator - maybe the variance in result comes only from the SPSA and the simulator does not need to be seeded.

Now I had believed there was some other deeper issue since BasicAer can fail internally with a NaN as one of the failure modes (the other being that the result was not a match to the expected one) @Cryoris was going to check about NaN.

Reply:
Hi Steve, 
I changed to `aqua_globals.random_seed = 123`, and I moved the code outside of the marked lines in the test file. Changing the sample code in the readme.md might confuse users that they think setting the random seed is necessary. Then, the SPSA loses its characteristic which is randomness. It's just my though. If you like the other way (setting the random seed in readme.md to make the code to be an exact copy of the sample in the readme.md), I will fix that.

Reply:
ah right... the test code failed with the error `ValueError: probabilities contain NaN` right now. This change can prevent providing infeasible answers, but not the NaN errors.
@woodsp-ibm

Reply:
Hopefully the seeding of SPSA is all that is needed around the infeasible result and the fact that we do not seed the simulator has no bearing. Maybe this is something that a user would expect - I would have hoped the problem was setup "better" and this randomness did not have such a large effect - for my liking it seems to fail too often. Though clearly given you say it still failed with NaN, which since it more seemed related to BasicAer, I guess we can hope that your change reduces the times it fails. I am not sure I would mark this as resolving #1193 though then as this only seems to partially alleviate the failures listed there.